### PR TITLE
[SPARK-13018][Docs] Replace example code in mllib-pmml-model-export.md using include_example

### PR DESCRIPTION
**Pull request by keypointt**
Replace example code in mllib-pmml-model-export.md using include_example
https://issues.apache.org/jira/browse/SPARK-13018

The example code in the user guide is embedded in the markdown and hence it is not easy to test. It would be nice to automatically test them. This JIRA is to discuss options to automate example code testing and see what we can do in Spark 1.6.

Goal is to move actual example code to spark/examples and test compilation in Jenkins builds. Then in the markdown, we can reference part of the code to show in the user guide. This requires adding a Jekyll tag that is similar to https://github.com/jekyll/jekyll/blob/master/lib/jekyll/tags/include.rb, e.g., called include_example.
`{% include_example scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala %}`
Jekyll will find `examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala` and pick code blocks marked "example" and replace code block in 
`{% highlight %}`
 in the markdown. 

See more sub-tasks in parent ticket: https://issues.apache.org/jira/browse/SPARK-11337

### CODE REVIEW

**Comment by mengxr** [comment]
ok to test

**Comment by mengxr** [comment]
cc @yinxusen

**Comment by mengxr** [comment]
@keypointt @yinxusen Could you start from this PR first and then other longer ones? @keypointt is new to Spark development. Starting from small ones could avoid unnecessary code rewrites. Thanks!

**Comment by yinxusen** [comment]
retest it please

File: examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// $example on$
+import org.apache.spark.mllib.clustering.KMeans
+import org.apache.spark.mllib.linalg.Vectors
+// $example off$
+
+object PMMLModelExportExample {
+
+  def main(args: Array[String]) {
+
+    val conf = new SparkConf().setAppName("PMMLModelExportExample").setMaster("local[*]")
```
Comment (yinxusen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala]: Don't set master here.

File: examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.apache.spark.{SparkConf, SparkContext}
+// $example on$
+import org.apache.spark.mllib.clustering.KMeans
+import org.apache.spark.mllib.linalg.Vectors
+// $example off$
+
+object PMMLModelExportExample {
+
+  def main(args: Array[String]) {
+
```
Comment (yinxusen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala]: remove the blank line

File: examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import org.apache.spark.{SparkConf, SparkContext}
+// $example on$
+import org.apache.spark.mllib.clustering.KMeans
+import org.apache.spark.mllib.linalg.Vectors
+// $example off$
+
+object PMMLModelExportExample {
+
+  def main(args: Array[String]) {
```
Comment (yinxusen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala]: `def main(args: Array[String]): Unit = {`

File: examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala (hunk 1 above)
Comment (keypointt) [inline examples/src/main/scala/org/apache/spark/examples/mllib/PMMLModelExportExample.scala]: ok, I've set master in other tickets besides this one, and also `def main(args: Array[String]): Unit
`

**Comment by yinxusen** [comment]
@keypointt I recommend adding the following code to the example, too. Then delete previous code from line 55-67. Meanwhile, we should move the line 53 ahead and rephrase it.

```scala
// Export the model to a String in PMML format
clusters.toPMML

// Export the model to a local file in PMML format
clusters.toPMML("/tmp/kmeans.xml")

// Export the model to a directory on a distributed file system in PMML format
clusters.toPMML(sc,"/tmp/kmeans")

// Export the model to the OutputStream in PMML format
clusters.toPMML(System.out)
```

@mengxr If we don't use https://github.com/apache/spark/pull/9430, I suggest to add those code together into a single file and rephrase those lines among those code snippets. Otherwise we get the following result:

screen shot 2016-02-13 at 7 33 28 pm

**Comment by yinxusen** [comment]
@mengxr LGTM

File: docs/mllib-pmml-model-export.md [hunk 4]
```diff
@@ -45,41 +45,12 @@ The table below outlines the `spark.mllib` models that can be exported to PMML a
 <div data-lang="scala" markdown="1">
 To export a supported `model` (see table above) to PMML, simply call `model.toPMML`.
 
+As well as exporting the PMML model to a String (`model.toPMML` as in the example above), you can export the PMML model to other formats.
```
Comment (mengxr) [inline docs/mllib-pmml-model-export.md]: minor: please wrap lines at 100 chars

**Comment by mengxr** [comment]
Merged into master. Thanks!

File: docs/mllib-pmml-model-export.md (hunk 4 above)
Comment (yinxusen) [inline docs/mllib-pmml-model-export.md]: Let's wrap it next time.

**Pull request closed without merging**